Projects/Application/Sources/Common/Network/Http/Http.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Created by Maxim Berezhnoy on 2019-02-12.
+// Licensed under the MIT license
+//
+//
+
+import Foundation
+
+// MARK: - RequestAdapter
+protocol RequestAdapter {
+    func adapt(urlRequest: URLRequest) -> URLRequest
```

Comment:
Shouldn't it use `Http.Request`?

Reply:
I was also considering that. However, I think using `URLRequest` better fits the purpose.
The `Http.Request` is a job, there is no point in using it during request adaptation.

Reply:
Added `Request` which serves as an entity wrapper around `URLRequest`. `RequestAdapter` now utilizes `Request` instead of `URLRequest`